plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -333,6 +334,24 @@ public void testNumericAggregationPushdown()
         }
     }
 
+    @Test
+    public void testCountDistinctWithStringTypes()
+    {
+        if (!(hasBehavior(SUPPORTS_CREATE_TABLE) && hasBehavior(SUPPORTS_INSERT))) {
+            throw new SkipException("Unable to CREATE TABLE to test count distinct");
+        }
+        String tableName = getSession().getSchema().orElseThrow() + ".count_distinct_varchars";
+        List<String> rows = ImmutableList.of("'a'", "'b'", "'A'", "'B'", "' a '", "'a'", "'b'", "' b '", "'a '", "'b '");
+        try (TestTable testTable = new TestTable(onRemoteDatabase(), tableName, "(t_varchar VARCHAR(5))", rows)) {
```

Comment:
You can set up data using `new TestTable(getQueryRunner()::execute,`

this way you don't need to try to be portable with the syntax (and it helps use unqualified table names)